### PR TITLE
yui0303 proposal: DoSM

### PR DESCRIPTION
**Pull request by yui0303**
---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [x] passing tests/CI

### CODE REVIEW

File: project/yui0303/README.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+\begin{bmatrix}
+  v_1 & v_2 & . & . & v_n
+\end{bmatrix}
+$$
+
+**prerequisite: A has n independent eigenvectors.**
+
+## Prospective Users
+
+The diagonal matrix has the property that all of its off-diagonal elements are zero. There are some prospective usages for related users are shown below.
```
Comment (yungyuc) [inline project/yui0303/README.md:139]: Eigenvalue problems are a big subject.  How large are the matrices you want to solve?  Is it dense or sparse?

File: project/yui0303/src/SystemArchitecture.jpg
Comment (yungyuc) [inline project/yui0303/src/SystemArchitecture.jpg:1]: I like your diagram.  I don't think the flow is entirely precise but it's a good start.

File: project/yui0303/README.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  - Start to coding the function of `CalEigenvalues()`
+- Week 2 (11/7): 
+  - Finish the the function of `CalEigenvalues()`
+- Week 3 (11/14): 
+  - Start to coding the function of `CalEigenvectors()`
+  - Try how to partition the calcuation of eignevalues and eignevectors with multi-threading or other accerlation methods.
+- Week 4 (11/21): 
+  - Keep tring the accerlation of matrix operation.
+- Week 5 (11/28): 
+  - Binding the c++ API with pybind11
```
Comment (yungyuc) [inline project/yui0303/README.md:208]: It's not easy to write tests before you have the binding ready.  Testing speeds up your development.  Please think about how to make yourself more productive.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [x] Add an entry in `project/README.rst` pointing to your proposal.
- [ ] Clarify the size of matrices and whether they are dense or sparse.

**Comment by yungyuc** [comment]
When you want me to review again, please leave a global comment.  Make sure the branch does not conflict.

**Comment by yungyuc** [comment]
@yui0303 Please do not resolve comments.  Resolution makes it harder for me to keep track of them.

File: hw3/.gitignore
Comment (yungyuc) [inline hw3/.gitignore:1]: Please remove unnecessary changesets from the git history.  Use rebase or a similar way.

File: project/yui0303/README.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+The user interface may base on [Vue](https://vuejs.org/) or [React](https://zh-hant.legacy.reactjs.org/).
+
+### Constraints
+
+- Size of input matrix: default any size
+  - The larger the matrix size, the higher the complexity. Thus, there may be a **TIMEOUT** mechanism during the computation.
+  - This mechanism depends on the parameters previously set in the environment building (using makefile or cmakefile).
+- The accerlation of the matrix operation is focus on **dense** matrix opertaion.
+
```
[the remark this answers is not in the document]
Comment (yungyuc) [reply project/yui0303/README.md:163]: You have not specify how large are the matrices you want to solve.  I do not insist you to include precise information in the proposal.  Please think about it and the information may enrich your presentation.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [ ] Remove unnecessary changesets.

File: hw3/.gitignore
Comment (yui0303) [reply hw3/.gitignore:1]: I have removed unnecessary changesets.

File: project/yui0303/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (yui0303) [reply project/yui0303/README.md:163]: Thank you for your reply, I now understand what the teacher meant.
My maximum input should be a 100x100 matrix. The reason is shown below:
The (time) complexity of finding eigenvalues and finding eigenvectors are O(n^4) and O(n^3) respectively.
Thus, when the matrix is 100x100, the complexity is approximately 10^8, and this is such a heavy burden for a personal computer.

Hope this reply can satisfy the teacher's question this time.

Note: O(n^4): The iteration loop is O(n) and the complexity of QR is O(n^3)

[the remark this answers is not in the document]
Comment (yungyuc) [reply project/yui0303/README.md:163]: It's a clear description.  Many other things may be derived from the analysis and I trust you will have a good start.

File: project/yui0303/README.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+- Week 5 (11/28): 
+  - Keep trying the accerlation of matrix operation.
+  - Implement a complete cmake file or makefile.
+- Week 6 (12/5):
+  - Prepare a presentaion skeleton
+  - Build the user interface and connect with the kernel function.
+- Week 7 (12/12):
+  - Maintain the system.
+  - Keep wrting the draft of final project.
+- Week 8 (12/19): Flextime.
```
[the remark this answers is not in the document]
Comment (yungyuc) [reply project/yui0303/README.md:222]: Don't mention it.  The more you tell me, the more I will feed back.  Keep doing it.

**Review by yungyuc — APPROVED** [review]
LGTM.  Let me know when you want it to be merged.

**Comment by yui0303** [comment]
> LGTM. Let me know when you want it to be merged.

Thank you for the review and the suggestion.  I want it to be merged now. 👍

**Comment by yungyuc** [comment]
@yui0303 I wonder if you have time for the meetup https://sciwork.dev/meetup/2023/1108-nycu this week.  I'd like to know your availability and progress update in the discord channel #nsd.  (Your repository is empty.)

**Pull request merged**